### DISCUSSION TITLE
How to increase the performance?

### DISCUSSION THREAD

**Discussion by kauderk** [Milkdown/milkdown#769] [Q&A]
I'm working on an node-graph-app ([point of reference](https://jerosoler.github.io/drawflow-vue3-example/)) and each node contains a milkdown editor ([point of reference](https://codesandbox.io/s/milkdown-i18n-2uzcvk?from-embed=&file=/index.html))

When there are more than 3 nodes: my browser slows down significantly, even more so when I open my Dev Tools
- While inspecting the head element I noticed that the library Emotion loads similar styles for each Editor
  - Can we use a global css file? For the nord theme for example.
- I want to give the users the ability to use the Menu Bar
  - Can we lazy load them? And make them visible only when the user interacts with them?
  - This [other tool](https://ckeditor.com/docs/ckeditor5/latest/examples/builds/inline-editor.html) gave me the idea.
- I expect the bootstrap/load times to increase the more Editors are created
  - Can we cash the Editor object or something similar?
- When I want to restore a node, to a previous state: I find my self destroying the component, updating the "defaultValue" then creating the same Editor.
  - Do we have access to any cleanup() or override() functions?
  
---
  Sorry if I'm being picky.

**Comment by Saul-Mirone** [comment]
1. I'm afraid we can't because the UI part is heavily depending on @emotion/css and they don't support that.
2. It's possible since 6.5. We support load plugin in runtime now.
3. I think the cache can be easily implemented in your app. It's not suitable for the library to handle something like this.
4. You can use `editor.action` and [replaceAll macro](https://milkdown.dev/macros#replaceall) to do that.
